pkg/auth/authenticator/password/ldappassword/ldap.go

```diff
@@ -164,10 +126,10 @@ func (a *Authenticator) getIdentity(username, password string) (authapi.UserIden
 
 	// Build list of attributes to retrieve
 	attrs := util.NewStringSet(a.options.URL.QueryAttribute)
-	attrs.Insert(a.options.AttributeEmail...)
-	attrs.Insert(a.options.AttributeName...)
-	attrs.Insert(a.options.AttributePreferredUsername...)
-	attrs.Insert(a.options.AttributeID...)
+	attrs.Insert(a.options.IdentityAttributeMapping.EmailAttribute...)
```

Comment:
This information is very tightly tied to the `IdentityFactory`, refactor to make the linkage very clear.